### PR TITLE
Add option for Telegram to use a thread id when posting notification

### PR DESCRIPTION
**Pull request by Techwolf12**
This also migrates the go library to the one from OvyFlash, since the current one seems to be no longer maintained:
https://github.com/go-telegram-bot-api/telegram-bot-api/issues/695

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -118,3 +118,5 @@ require (
 )
 
 replace github.com/prometheus/client_golang => github.com/prometheus/client_golang v1.14.0
+
+replace github.com/go-telegram-bot-api/telegram-bot-api/v5 => github.com/OvyFlash/telegram-bot-api/v5 v5.0.0-20240108230938-63e5c59035bf
```
[the remark this answers is not in the document]
Comment (Techwolf12) [reply go.mod:122]: It's a maintained fork that still uses `github.com/go-telegram-bot-api/telegram-bot-api/v5` as the package

File: pkg/services/telegram.go [hunk 2]
```diff
@@ -35,6 +37,14 @@ func (s telegramService) Send(notification Notification, dest Destination) error
 		msg := tgbotapi.NewMessage(chatID, notification.Message)
 		msg.ParseMode = "Markdown"
 
+		if len(chatChannel) > 1 {
```
[the remark this answers is not in the document]
Comment (Techwolf12) [reply pkg/services/telegram.go:37]: Written some tests!

File: go.mod (hunk 1 above)
[the remark this answers is not in the document]
Comment (pasha-codefresh) [reply go.mod:122]: yeah, i understand it, but you can just change dependency and not use replace option , no ?

File: pkg/services/telegram.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (pasha-codefresh) [reply pkg/services/telegram.go:37]: Thank you!

File: go.mod (hunk 1 above)
[the remark this answers is not in the document]
Comment (Techwolf12) [reply go.mod:122]:
```
module declares its path as: [github.com/go-telegram-bot-api/telegram-bot-api/v5](http://github.com/go-telegram-bot-api/telegram-bot-api/v5)
                but was required as: [github.com/OvyFlash/telegram-bot-api/v5](http://github.com/OvyFlash/telegram-bot-api/v5)
```

I don't see an option to do it like this, considering it's a fork that still defines the old module

[the remark this answers is not in the document]
Comment (pasha-codefresh) [reply go.mod:122]: Understand, let me double check it in my fork , i will try finalize it today

**Review by pasha-codefresh — APPROVED** [review]
LGTM, thank you

**Pull request merged**